src/amo/server/webpack-assets-fonts.js

```diff
@@ -0,0 +1,36 @@
+import fs from 'fs';
+
+const pluginName = 'WebpackAssetsFontsPlugin';
+const webpackAssetsFileName = 'webpack-assets.json';
+// This is a webpack plugin to add .woff2 fonts in webpack-assets.json that
+// webpack-isomorphic-tools generates. We can then use `assets` to reference
+// these fonts in server side rendered code to preload some font subsets, like
+// we can in development mode.
+
+export default class WebpackAssetsFontsPlugin {
+  apply(compiler) {
+    compiler.hooks.done.tap(pluginName, (stats) => {
+      const subsetFonts = {};
+      const { assets, publicPath } = stats.toJson();
+
+      try {
+        assets.forEach((asset) => {
+          const {
+            name,
+            info: { sourceFilename },
+          } = asset;
+
+          if (name.endsWith('.woff2')) {
+            subsetFonts[`./${sourceFilename}`] = `${publicPath}${name}`;
+          }
+        });
+
+        const data = JSON.parse(fs.readFileSync(webpackAssetsFileName));
+        data.assets = Object.assign(data.assets, subsetFonts);
```

Comment:
```suggestion
        data.assets = { ...data.assets, ...subsetFonts };
```

`Object.assign()` modifies the first argument so I think `data.assets` is updated _before_ being reassigned, which isn't what you want. In the past, we had to do `Object.assign({}, data.assets, subsetFonts)` but now we can use spread.

Reply:
thanks, I keep forgetting about spread...